### ISSUE TITLE
Date posted cannot be prior to date user joined

### ISSUE THREAD

**Issue by stevelacey** [jschementi/flickr_photostream_sorter#3]
I have a few that change every time I run the script – so something clearly isn't quite right, I'm guessing it's failing to set the taken at date properly.

https://www.flickr.com/photos/stevelacey

```
-----> Fetching all photos
       Fetching page 1 out of 1
       Fetching page 2 out of 5
       Fetching page 3 out of 5
       Fetching page 4 out of 5
       Fetching page 5 out of 5
-----> Updating dates
       Skipping "17105835799": dates match
       ...
       Skipping "3264097106": dates match
       Updating "3255450741": change date posted from 2009-02-05 17:51:39 to 2008-02-05 17:35:24
       Updating "3256105040": change date posted from 2009-02-05 16:23:49 to 2008-02-05 16:12:00
       Updating "3255248911": change date posted from 2009-02-05 16:10:46 to 2008-02-05 13:35:29
       Updating "3256079086": change date posted from 2009-02-05 16:10:32 to 2008-02-05 13:34:32
       Updating "3256078684": change date posted from 2009-02-05 16:10:18 to 2008-02-05 13:34:12
       Updating "3255247489": change date posted from 2009-02-05 16:10:05 to 2008-02-05 13:34:03
       Updating "3256077624": change date posted from 2009-02-05 16:09:50 to 2008-02-05 13:32:50
       Updating "3255246491": change date posted from 2009-02-05 16:09:39 to 2008-02-05 13:32:35
       Updating "3256076710": change date posted from 2009-02-05 16:09:24 to 2008-02-05 13:27:09
       Updating "3255245215": change date posted from 2009-02-05 16:09:04 to 2008-02-05 13:26:50
       Updating "3256075432": change date posted from 2009-02-05 16:08:48 to 2008-02-05 13:24:46
       Updating "3256074826": change date posted from 2009-02-05 16:08:32 to 2008-02-05 13:23:38
       Updating "3255243467": change date posted from 2009-02-05 16:08:20 to 2008-02-05 13:22:00
       Updating "3255243091": change date posted from 2009-02-05 16:08:10 to 2008-02-05 13:21:55
       Updating "3256073596": change date posted from 2009-02-05 16:07:56 to 2008-02-05 13:21:32
       Updating "3255242139": change date posted from 2009-02-05 16:07:42 to 2008-02-05 13:21:17
       Updating "3256072716": change date posted from 2009-02-05 16:07:31 to 2008-02-05 11:15:22
       Updating "3255241195": change date posted from 2009-02-05 16:07:16 to 2008-02-05 11:15:00
       Updating "3255240829": change date posted from 2009-02-05 16:07:05 to 2008-02-05 11:12:17
       Updating "3255240425": change date posted from 2009-02-05 16:06:54 to 2008-02-05 11:12:06
       Updating "3255239965": change date posted from 2009-02-05 16:06:41 to 2008-02-05 11:11:55
       Updating "3255239571": change date posted from 2009-02-05 16:06:30 to 2008-02-05 11:11:48
       Updating "3256070052": change date posted from 2009-02-05 16:06:16 to 2008-02-05 02:25:57
       Updating "3255238433": change date posted from 2009-02-05 16:05:59 to 2008-02-05 02:25:41
       Updating "3256068592": change date posted from 2009-02-05 16:05:34 to 2008-02-05 02:25:28
       Updating "3256067970": change date posted from 2009-02-05 16:05:15 to 2008-02-05 02:24:51
       Updating "3256067280": change date posted from 2009-02-05 16:04:54 to 2008-02-02 20:44:18
       Updating "3256066818": change date posted from 2009-02-05 16:04:39 to 2008-02-02 20:43:43
       Updating "3256066290": change date posted from 2009-02-05 16:04:23 to 2008-02-05 13:38:56
       Skipping "3255947604": dates match
       Skipping "3255116461": dates match
       ...
       Skipping "3068344530": dates match
       Updating "3059772622": change date posted from 2008-11-25 20:42:27 to 2007-12-15 16:12:38
-----> Done!
```

**Comment by stevelacey** [comment]
Just to make it clear:

```
[~]$ flickr_photostream_sorter | grep 3256079086                                                                                                                                                                                             
       Updating "3256079086": change date posted from 2009-02-05 16:10:32 to 2008-02-05 13:34:32
[~]$ flickr_photostream_sorter | grep 3256079086                                                                                                                                                                                             
       Updating "3256079086": change date posted from 2009-02-05 16:10:32 to 2008-02-05 13:34:32
```

**Comment by madsbk** [comment]
I have the same problem. In my case, it is because you are not allowed to change the "date posted" to a date before you joined Flickr

**Comment by stevelacey** [comment]
@madsbk bingo, same with me, I joined in 2008 – hilarious. I guess a neat addition to this would be to set older photos to the day you joined then :|

**Comment by madsbk** [comment]
@stevelacey I implemented just that in #6 :)

**Comment by stevelacey** [comment]
@madsbk slick, the +seconds solution sounds great
